docs/src/main/sphinx/admin/properties-node-scheduler.rst

```diff
@@ -30,21 +30,35 @@ due to splits not being balanced across workers. Ideally, it should be set
 such that there is always at least one split waiting to be processed, but
 not higher.
 
-``node-scheduler.max-pending-splits-per-task``
+``node-scheduler.min-pending-splits-per-task``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
 * **Type:** :ref:`prop-type-integer`
 * **Default value:** ``10``
 
-The number of outstanding splits with the standard split weight that can be
-queued for each worker node for a single stage of a query, even when the
-node is already at the limit for total number of splits. Allowing a minimum
+The minimum number of splits with the standard split weight guaranteed to be scheduled
+on a node for a single task given the task has remaining splits to process. Allowing a minimum
 number of splits per stage is required to prevent starvation and deadlocks.
 
-This value must be smaller than ``node-scheduler.max-splits-per-node``,
-is usually increased for the same reasons, and has similar drawbacks
+This value must be smaller or equal than ``max-adjusted-pending-splits-per-task`` and
+``node-scheduler.max-splits-per-node``, is usually increased for the same reasons,
+and has similar drawbacks if set too high.
+
+``node-scheduler.max-adjusted-pending-splits-per-task``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Default value:** ``500``
+
+The maximum number that ``min-pending-splits-per-task`` could be adjusted during scheduling splits
```

Comment:
adjusting of `min-pending-splits-per-task` seems confusing. Maybe:
```
The maximum number of outstanding splits with the standard split weight that can be
queued for each worker node for a single stage of a query, even when the
node is already at the limit for total number of splits. Split queue size is adjusted
dynamically during split scheduling and cannot exceed ``node-scheduler.max-adjusted-pending-splits-per-task``.
Split queue size per task will be adjusted upward if node processes splits
faster than it receives them.
```

@mosabua @colebow could you check this one?